File: fastapi/app/controller/AccountController.py
from fastapi import APIRouter
from app.model.AccountDTO import AccountDTO
from app.service.AccountService import AccountService
from typing import Dict
from app.payload.response.ApiResponse import ApiResponse
from http import HTTPStatus

router = APIRouter(
    prefix="/api/account"
)

@router.post("/signup", response_model=ApiResponse)
def signUpAccount(data: AccountDTO):
    retrun_msg = AccountService.signUpAccount(data)
    return ApiResponse.ok(code=HTTPStatus.OK.value, message=HTTPStatus.OK.phrase, data=retrun_msg)

File: springboot/src/main/java/com/hypermedicus/controller/AccountController.java
package com.hypermedicus.controller;

import com.hypermedicus.model.AccountDTO;
import com.hypermedicus.payload.response.AccountResponse;
import com.hypermedicus.payload.response.ApiResponse;
import com.hypermedicus.service.AccountService;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/account/signup")
public class AccountController {

    private final AccountService accountService;

    public AccountController(AccountService accountService) {
        this.accountService = accountService;
    }

    @PostMapping
    public ApiResponse<AccountResponse> signUpAccount(@RequestBody AccountDTO dto) {
        AccountResponse accountResponse = accountService.signUpAccount(dto);
        return ApiResponse.ok(accountResponse);
    }
}
